### PR TITLE
Support JVM on M1

### PR DESCRIPTION
**Pull request by nhachicha**
Build macos JVM dylib on an M1 node

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -121,6 +121,20 @@ pipeline {
                               build_jvm_linux()
                           }
                       }
+
+                      stage('build_osx_arm64') {
```
Comment (cmelchior) [inline Jenkinsfile:125]: Didn't we build the macosX64 binaries before? 🤔

**Review by cmelchior — COMMENTED** [review]
Looks good, but aren't we missing support for intel macos in a few places? At least I don't see it referenced in some of the build steps.

File: Jenkinsfile [hunk 2]
```diff
@@ -326,6 +340,7 @@ def runBuild() {
     if (shouldBuildJvmABIs()) {
         unstash name: 'linux_so_file'
         unstash name: 'win_dll'
+        unstash name: 'osx_arm64_so_file'
```
[the remark this answers is not in the document]
Comment (nhachicha) [reply Jenkinsfile:343]: No, the macos `x86_64` is already built in the main node https://github.com/realm/realm-kotlin/blob/master/packages/cinterop/build.gradle.kts#L347-L366 

Here are all the artifacts after a successful build https://ci.realm.io/blue/organizations/jenkins/realm%2Frealm-kotlin/detail/PR-651/12/artifacts

File: Jenkinsfile (hunk 1 above)
Comment (nhachicha) [reply Jenkinsfile:125]: Ditto

File: Jenkinsfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmelchior) [reply Jenkinsfile:343]: Ah great, I was bad at searching 🙈

**Review by cmelchior — APPROVED** [review]

**Review by rorbech — APPROVED** [review]

**Comment by nhachicha** [comment]
@cmelchior @rorbech FYI I simplified the implementation to build a universal macos shared lib (containing both arm and intel) we no longer requires a separate node to build the M1

**Review by cmelchior — APPROVED** [review]
Nice 💯

**Pull request merged**